### PR TITLE
Small cleanups for filterset

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Remove unnecessary Options from regexp and strict filters
* Use slice for regexp when iterating (faster than iterating over map)
* Fix bug in de-dup logic for regexp (`flag.CommandLine.Name()` was used instead the filter value)

### CODE REVIEW

**Comment by ccaraman** [comment]
@bogdandrutu - please pull in master to include the latest test changes for BatchProcessor.

File: internal/processor/filterset/config.go [hunk 1]
```diff
@@ -45,10 +47,10 @@ type Config struct {
...
 	switch cfg.MatchType {
 	case Regexp:
-		return regexp.CreateRegexpFilterSet(filters, cfg.RegexpConfig)
+		return regexp.NewFilterSet(filters, cfg.RegexpConfig)
 	case Strict:
 		// Strict FilterSets do not have any extra configuration options, so call the constructor directly.
-		return strict.NewStrictFilterSet(filters)
+		return strict.NewFilterSet(filters)
 	default:
 		return nil, fmt.Errorf("unrecognized match_type: '%v', valid types are: %v", cfg.MatchType, validMatchTypes)
```
Comment (james-bebbington) [inline internal/processor/filterset/config.go:55]:
```suggestion
		return nil, fmt.Errorf("unrecognized %v: '%v', valid types are: %v", MatchTypeFieldName, cfg.MatchType, validMatchTypes)
```

Comment (bogdandrutu) [reply internal/processor/filterset/config.go]: Done.

**Comment by tigrannajaryan** [comment]
@bogdandrutu can you please add PR description / commit message to help explain the purpose of the cleanup?

**Comment by bogdandrutu** [comment]
@tigrannajaryan done

**Pull request merged**